lib/Auth/Source/LdapMulti.php

```diff
@@ -100,24 +108,24 @@ public function __construct(array $info, array $config)
      *
      * @param string $username  The username the user wrote.
      * @param string $password  The password the user wrote.
-     * @param string $organization  The organization the user chose.
      * @return array  Associative array with the users attributes.
      */
-    protected function login(string $username, string $password, string $organization, array $sasl_args = null): array
+    protected function login(string $username, string $password, string $organization): array
     {
-        if (!array_key_exists($organization, $this->ldapOrgs)) {
-            // The user has selected an organization which doesn't exist anymore.
-            Logger::warning('Authentication source ' . var_export($this->authId, true) .
-                ': Organization seems to have disappeared while the user logged in.' .
-                ' Organization was ' . var_export($organization, true));
-            throw new Error\Error('WRONGUSERPASS');
-        }
-
         if ($this->includeOrgInUsername) {
             $username = $username . '@' . $organization;
         }
 
-        return $this->ldapOrgs[$organization]->login($username, $password, $sasl_args);
+        $authsource = $this->mapping[$organization]['authsource'];
+        $sourceConfig = $this->ldapOrgs[$organization];
+
+        $ldap = new class (['AuthId' => $authsource], $sourceConfig->toArray()) extends Ldap {
```

Comment:
This is a bit of a dirty hack, but there is currently no way to delegate authentication to another authsource.
The positive side is though, that we can use references to other LDAP-authsources in the config, instead of importing entire LDAP-configurations for multiple organizations.